atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -390,6 +388,11 @@ private enum State {
         this.preCommitRequirementValidator = new DefaultPreCommitRequirementValidator(
                 preCommitCondition, transactionOutcomeMetrics, immutableTimestampLockManager);
         this.commitTimestampLoader = commitTimestampLoader;
+        this.readSentinelHandler = new ReadSentinelHandler(
+                transactionKeyValueService,
+                transactionService,
+                readSentinelBehavior,
+                new DefaultOrphanedSentinelDeleter(sweepStrategyManager::get, deleteExecutor));
```

Comment:
more of a question than a comment: along these refactors we've been doing a bunch of object instantiations here, rather the usual preferred method of dependency injection through the constructor. I assume that it's intentional so we don't break the constructor and just introduce a new one where we pass all of these once we've stabilized the interface at the end of the refactors?

Reply:
Yep, that's right. Not the long-term goal to have too much construction logic here, indeed.